src/dir.proj

```diff
@@ -0,0 +1,292 @@
+
+    <Exec Command="$(DotnetToolCommand) publish --output $(runtimeGraphGeneratorOutput) $(runtimeGraphGeneratorProject) "
+          EnvironmentVariables="NUGET_PACKAGES=$(PackagesDir)" />
+
+    <PropertyGroup>
+      <RuntimeGraphGeneratorExe>$(runtimeGraphGeneratorOutput)/$(runtimeGraphGeneratorName)$(ExeSuffix)</RuntimeGraphGeneratorExe>
+      <DepsJson>$(SharedFrameworkNameAndVersionRoot)\$(SharedFrameworkName).deps.json</DepsJson>
+
+    </PropertyGroup>
+
+    <Exec Command="$(RuntimeGraphGeneratorExe) --project $(SharedFrameworkSourceRoot) --deps $(DepsJson) $(runtimeGraphGeneratorRuntime)"
+          EnvironmentVariables="NUGET_PACKAGES=$(PackagesDir)" />
+  </Target>
+
+  <Target Name="RestoreLockedCoreHost">
+    <ItemGroup>
+      <ProjJsonLines Include='{' />
+      <ProjJsonLines Include='  "dependencies": {' />
+      <ProjJsonLines Include='    "Microsoft.NETCore.DotNetHostResolver"  :  "$(HostVersion)"' />
+      <ProjJsonLines Include='  },' />
+      <ProjJsonLines Include='  "frameworks": {' />
+      <ProjJsonLines Include='    "$(Framework)": {}' />
+      <ProjJsonLines Include='  },' />
+      <ProjJsonLines Include='  "runtimes": {' />
+      <ProjJsonLines Include='    "$(PackageTargetRid)": {}' />
+      <ProjJsonLines Include='  }' />
+      <ProjJsonLines Include='}' />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <JsonDir>$(IntermediateOutputRootPath)lockedHostTemp</JsonDir>
+    </PropertyGroup>
+
+    <MakeDir Condition="!Exists('$(JsonDir)')" Directories="$(JsonDir)" />
+
+    <WriteLinesToFile
+      File="$(JsonDir)\project.json"
+      Lines="@(ProjJsonLines)"
+      Overwrite="true" />
+    
+    <PropertyGroup>
+      <RestoreArgs>--source https:%2F%2Fdotnet.myget.org/F/dotnet-core/api/v3/index.json --fallbacksource $(CoreHostOutputDir) --packages "$(PackagesDir.TrimEnd('/\'.ToCharArray()))"</RestoreArgs>
+    </PropertyGroup>  
+
+    <Exec Command="$(DotnetToolCommand) restore --verbosity verbose $(RestoreArgs)"
+          WorkingDirectory="$(JsonDir)"
+          EnvironmentVariables="NUGET_PACKAGES=$(PackagesDir)" />
+
+    <PropertyGroup>
+      <PublishArgs>--output $(CoreHostLockedDir) --no-build -r $(PackageTargetRid)</PublishArgs>
+    </PropertyGroup>
+
+    <RemoveDir Directories="$(CoreHostLockedDir)" />
+
+    <Exec Command="$(DotnetToolCommand) publish $(PublishArgs)"
+          WorkingDirectory="$(JsonDir)"
+          EnvironmentVariables="NUGET_PACKAGES=$(PackagesDir)" />
+  </Target>
+
+  <Target Name="GenerateSharedFrameworkProject">
+
+    <MakeDir Condition="!Exists('$(IntermediateOutputRootPath)sharedFramework')" 
+             Directories="$(IntermediateOutputRootPath)sharedFramework;$(SharedFrameworkSourceRoot)" />
+
+    <ItemGroup>
+      <SharedFrameworkFiles Include="sharedFramework\**\*.cs" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(SharedFrameworkFiles)" DestinationFolder="$(SharedFrameworkSourceRoot)"/>
+
+    <ItemGroup>
+      <JsonLines Include='{' />
```

Comment:
Why are we generating this file?  We already have a project.json.template file:

[https://github.com/dotnet/core-setup/blob/master/src/sharedframework/framework/project.json.template](https://github.com/dotnet/core-setup/blob/master/src/sharedframework/framework/project.json.template)

Let's just use that, and replace the 3 variables in it, instead of generating this whole file in MSBuild code.